### ISSUE TITLE
Doesn't support xml with namespaces

### ISSUE THREAD

**Issue by vicp-iq** [IngoGege/xXMLConfigFile#5]
I'm trying to update a file that is shaped like this, and there is no way to apply the namespace information.

```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
	<PropertyGroup>
		<Something>Value</Something>
	</PropertyGroup>
</Project>
```

**Comment by IngoGege** [comment]
Hi @vicp-iq , just to make sure I understood you correct: You want to modify the xmmlns namespace?

**Comment by vicp-iq** [comment]
No, I want to add new elements to the property group, but https://github.com/IngoGege/xXMLConfigFile/blob/master/Misc/xXMLConfigFileCommonFunctions.psm1#L20 isn't using the overload that accepts a namespace manager.

**Comment by IngoGege** [comment]
That's what is written in the section known issues https://github.com/IngoGege/xXMLConfigFile#known-issues . I'm working on it, but I haven't found a good way to add nodes as there are multiple kinds of nodes...

**Comment by vicp-iq** [comment]
I see that it says: `"You can add additional elements with the name 'add key=... value=...', but not with a different name like 'remove key=... value=...'.`

I am trying to add additional elements with the same name, so it seems like this says that's a supported operation.  The only problem is that the namespace at the top makes the operation fail.

**Comment by IngoGege** [comment]
Can you please write, how it should look like, afterwards?

**Comment by vicp-iq** [comment]
Oh, my previous comment makes less sense when I refresh my view of the file.  Here's my objective:

Before:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
	<PropertyGroup>
		<ElementA>ValueA</ElementA>
	</PropertyGroup>
</Project>
```
After:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
	<PropertyGroup>
		<ElementA>ValueA</ElementA>
		<ElementB>ValueB</ElementB>
	</PropertyGroup>
</Project>
```

**Comment by IngoGege** [comment]
Yeah, my statement might be confusing, but this is exactly the issue: Element B is a new node with a text node Value. You could add a new node Element A with a different text node.
But I might have a solution for this. Give me a few days. I have to extend also the other functions.

**Comment by vicp-iq** [comment]
Ahh, thanks for the clarification!  And thank you for the assistance!

**Comment by IngoGege** [comment]
Hi @vicp-iq , I just opened another issue as you experience 2 issues:
the one you just found, which is caused that the resource is using SelectSingleNode without specifying any namespace. This works if you have no namespace defined, but like in your example no node will be found.
https://msdn.microsoft.com/en-us/library/h0hw012b(v=vs.110).aspx#Anchor_2
The second issue is as described the fact that the resource caccont add new nodes.

I will add another parameter, where you can specify a namespace in order to find the nodes.
I'm currently struggling how to handle the XPath in such a case.
Ciao,
Ingo

**Comment by vicp-iq** [comment]
Thanks for the update!

**Comment by IngoGege** [comment]
Hi @vicp-iq , I've added the support for XML namespaces. As next I will fix issue #6

**Comment by vicp-iq** [comment]
Sweet thanks.
